### PR TITLE
Add PySide GUI to make importador more user friendly

### PR DESCRIPTION
**Pull request by tupaschoal**
# tl;dr

I used it in its current state to import over 1100 games I had on BGG over to Ludopedia. I tried doing them all at once, but the script failed (probably because I was shut down by Ludopedia for spamming). Anyway, I then proceed to import ranges of 6mo and everything went smoothly, not a single play failed to be posted. 

Here's how it looks in action:
demoImportador

PS: I made the e-mail hidden like a password to avoid exposing myself on the internet :D
PPS: GIF said no games were posted just because I didn't want to duplicate what I already have

# Summary of user visible changes

- Only way to interact is now with the GUI
- End date does not update to the start date automatically, but it is already initialized as so (I felt like this was annoying when using the GUI)
- User can preview the if the user list mapping has been loaded correctly before importing
- Messages with ongoing operations are printed with timestamps, for better sense of progress
- When a game search does not find one with the same name and published year, it now allows the user to pick one of the options found. **This makes it safer than automatically matching to the first**
- When a game search returns no results, user can search for another string to try finding a match.
- Changed icon to a homebrewed one which is an overlap of bgg and ludopedia's
- When adding an "Observação" to a player on a play, it now also adds color, position and whether the player was new or not (drawn from the BGG logged play)

# Summary of code-related changes

The diff won't be very friendly, sorry :(
- Now uses Pyside6 (Qt for Python 6) for drawing GUIs
- Main GUI controls all the actions and logging, all of the communication with BGG and Ludopedia is done on separate (worker) threads to avoid hanging the GUI
- Upon failing to login or post, error messages in orangeish are posted to the app log.
- Added a pyinstaller spec file to make a OS independent configuration file for generating the releases. I tested this on Windows and it works fine, but I didn't confirm on MacOS.

# Ideas for the future

I have collected some ideas for expanding and/or making it better in the future. I was going to implement those, but I felt like I needed to get this "first" version out of the window before I moved on. I might open issue to track those, if I don't feel like doing them now. These ideas are:
- Allow user to map BGG and Ludopedia users from GUI (or point to the file from the GUI)
- Validate BGG and Ludopedia users beforehand
- Get ludopedia user ids automatically, so that app user can input their login name instead
- Preview plays and games to be imported on a list, so you can check what to include or not before actually posting them
- Require all game mappings to be confirmed (disables automatic mapping via search + published year)

# Screenshots

**Main screen:**
image

**User mapping list:**
image

**Choose alternative Dialog:**
image

**Search Alternative Dialog:**
image

**Search Results Dialog:**
image

### CODE REVIEW

**Comment by tupaschoal** [comment]
@dambros ?

**Comment by dambros** [comment]
@tupaschoal I am sorry for the delay on reviewing this, I should be able to get to it this weekend.

Judging by the PR description and images, this looks amazing. It is something I always wanted since the initial idea, so just for this reason alone, thanks!

**Review by dambros — APPROVED** [review]
This is overall one of the best PR I had the chance of reviewing. Great work :)

**Comment by tupaschoal** [comment]
Thanks for the compliment :)
